src/find-options/FindOptionsUtils.ts

```diff
@@ -94,6 +96,11 @@ export class FindOptionsUtils {
             return qb;
 
         // apply all options from FindOptions
+        if (options.select) {
+            qb.select(options.select.map(selection => qb.alias + "." + selection));
+            qb.partialSelect = true;
```

Comment:
I couldn't figure out a way to differentiate this select and the regular ones that are done with the query builder. Using virtual columns makes other query builder tests to break because a primary column is inserted into the sql. I don't like this, and I'm hoping you can think of something better.

Reply:
I'm rude? Sorry if I am, I don't want to be so haha. I just wanted to say that this code can be better and must be redesigned (you know it too, right)

Reply:
Okay that's good. I know the flag isn't right. I definitely do not want it and would prefer if it is removed. I just wanted to demonstrate the issue to you.

If I only check if there have been selects, it breaks the other query builder tests that check the sql output. I'm willing to wait until you're back from vacation since you have a better understanding of the code. There might be something obvious that would solve this that I've missed. 

Enjoy your holiday.